rados/rados_write_op_assert_version_test.go

```diff
@@ -0,0 +1,42 @@
+//go:build ceph_preview
+// +build ceph_preview
+
+package rados
+
+import (
+	"github.com/stretchr/testify/assert"
+)
+
+func (suite *RadosTestSuite) TestWriteOpAssertVersion() {
+	suite.SetupConnection()
+	ta := assert.New(suite.T())
+
+	var (
+		oid = "TestWriteOpAssertVersion"
+		err error
+	)
+
+	// Create an object.
+	op1 := CreateWriteOp()
+	defer op1.Release()
+	op1.Create(CreateIdempotent)
+	err = op1.Operate(suite.ioctx, oid, OperationNoFlag)
+	ta.NoError(err)
+
+	ver, err := suite.ioctx.GetLastVersion()
+
+	// Write with version assert. It should succeed.
+	op2 := CreateWriteOp()
+	defer op2.Release()
+	op2.AssertVersion(ver)
+	err = op2.Operate(suite.ioctx, oid, OperationNoFlag)
+	ta.NoError(err)
+
+	// Write with version assert, but modify the version first.
+	// It should fail.
+	op3 := CreateWriteOp()
+	defer op3.Release()
+	op3.AssertVersion(ver + 1)
```

Comment:
I'm not sure how rados versions work exactly, but I'm curious if it could actually be ver +1 in some cases because the previous action is a write operation?

Reply:
I've updated the test to retrieve the object version again and then compare against that +1.